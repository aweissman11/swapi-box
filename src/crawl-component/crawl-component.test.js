import React from 'react';
import { shallow } from 'enzyme';

import CrawlComponent from './crawl-component';

describe('CrawlComponent', () => {
  it('should match the snapshot', () => {
    let movie = { title: 'StarTrek', date: 'December 31 1999', opening: 'Space: the final frontier.  These are the voyages of the starship Enterprise.  Its five-year mission: to explore strange new worlds. To seek out new life and new civilizations.  To boldly go where no man has gone before!', somethingElse: 'info that will later be removed from this object'};
    const wrapper = shallow(<CrawlComponent movie={movie}/>);
    expect(wrapper).toMatchSnapshot();
  });
});