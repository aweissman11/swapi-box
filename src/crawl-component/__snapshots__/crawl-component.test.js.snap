// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CrawlComponent should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MyCrawlComponent
    movie={
      Object {
        "date": "December 31 1999",
        "opening": "Space: the final frontier.  These are the voyages of the starship Enterprise.  Its five-year mission: to explore strange new worlds. To seek out new life and new civilizations.  To boldly go where no man has gone before!",
        "somethingElse": "info that will later be removed from this object",
        "title": "StarTrek",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "subTitle": "StarTrek",
      "text": <span>
        <span>
          Space: the final frontier.  These are the voyages of the starship Enterprise.  Its five-year mission: to explore strange new worlds. To seek out new life and new civilizations.  To boldly go where no man has gone before!
        </span>
        <br />
        <span>
          Released: 
          December 31 1999
        </span>
      </span>,
      "title": "STAR WARS",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "subTitle": "StarTrek",
        "text": <span>
          <span>
            Space: the final frontier.  These are the voyages of the starship Enterprise.  Its five-year mission: to explore strange new worlds. To seek out new life and new civilizations.  To boldly go where no man has gone before!
          </span>
          <br />
          <span>
            Released: 
            December 31 1999
          </span>
        </span>,
        "title": "STAR WARS",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
